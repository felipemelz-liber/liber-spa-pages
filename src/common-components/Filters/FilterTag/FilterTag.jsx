import React from 'react';
import PropTypes from 'prop-types';
import {
  FilterTag as Tag,
  SelectedLabelContainer,
  SelectedLabelOption,
  SelectedSwitch,
} from './FilterTag.styles';

const isSelected = list => {
  if (list) {
    return list.filter(item => Boolean(item)).length !== 0;
  }
  return false;
};

const format = list => {
  return list.join(' ');
};

const FilterTag = ({ selected, formatSelected, label, onClear, show, ...props }) => {
  const [hover, setHover] = React.useState(false);
  let tagProps;
  if (isSelected(selected)) {
    const nodeLabel = (
      <SelectedLabelContainer>
        <SelectedSwitch switch={hover || show}>
          <SelectedLabelOption>{formatSelected(selected)}</SelectedLabelOption>
          <SelectedLabelOption>{label}</SelectedLabelOption>
        </SelectedSwitch>
      </SelectedLabelContainer>
    );
    tagProps = {
      label: nodeLabel,
      closable: true,
      onClose: onClear,
      onMouseEnter: () => setHover(true),
      onMouseLeave: () => setHover(false),
      gray: !show,
      show,
      ...props,
    };
  } else {
    tagProps = { label, show, ...props };
  }
  return <Tag {...tagProps} />;
};

FilterTag.propTypes = {
  selected: PropTypes.arrayOf(PropTypes.any),
  formatSelected: PropTypes.func,
  onClear: PropTypes.func,
  label: PropTypes.string,
  show: PropTypes.bool,
};

FilterTag.defaultProps = {
  selected: [],
  formatSelected: format,
  onClear: () => {},
  show: false,
  label: '',
};

export default FilterTag;
