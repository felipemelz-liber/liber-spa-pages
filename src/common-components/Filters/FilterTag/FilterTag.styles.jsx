import styled, { css } from 'styled-components';
import { ModalTag } from 'liber-components';

const isGray = ({ gray }) => {
  if (gray)
    return css`
      background-color: #f4f5f7;
      border-color: #f4f5f7;
    `;
  return null;
};

export const FilterTag = styled(ModalTag)`
  font-size: 16px;
  min-height: 36px;
  border-radius: 4px;
  font-weight: 400;
  ${isGray}
`;

export const SelectedLabelContainer = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
`;

export const SelectedSwitch = styled.div`
  height: 52px;
  width: 100%;
  padding-top: 6px;
  display: flex;
  flex-direction: column;
  transform: ${props => (props.switch ? 'translateY(-26px)' : 'translateY(0px)')};
  transition: transform 250ms ease-in-out;
`;

export const SelectedLabelOption = styled.span`
  width: 100%;
  height: 26px;
  display: flex;
  justify-content: center;
  align-items: center;
`;
