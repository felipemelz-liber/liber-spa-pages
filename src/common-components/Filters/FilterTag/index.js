export { default } from './FilterTag.jsx';
