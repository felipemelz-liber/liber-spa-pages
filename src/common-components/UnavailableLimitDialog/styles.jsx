import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
