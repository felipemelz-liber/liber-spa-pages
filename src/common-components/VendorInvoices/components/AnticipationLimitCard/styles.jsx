import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
