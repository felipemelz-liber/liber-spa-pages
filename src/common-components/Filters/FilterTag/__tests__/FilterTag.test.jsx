import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import 'jest-styled-components';
import FilterTag from '../FilterTag';
import { FilterTag as Tag } from '../FilterTag.styles';

describe('Testing FilterTag Component', () => {
  configure({ adapter: new Adapter() });
  let props;
  beforeEach(() => {
    props = {
      selected: [],
      onClear: sinon.spy(),
      show: false,
      label: 'label',
    };
  });
  it('should render correctly without props', () => {
    const wrapper = shallow(<FilterTag />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with props', () => {
    const wrapper = shallow(<FilterTag {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly open', () => {
    props.show = true;
    const wrapper = shallow(<FilterTag {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with selected', () => {
    props.selected = ['selected value'];
    const wrapper = shallow(<FilterTag {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with selected wile open', () => {
    props.selected = ['selected value'];
    props.show = true;
    const wrapper = shallow(<FilterTag {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with custom seleted Formatter', () => {
    props.selected = [1, 2, 3];
    props.formatSelected = list => <div>{list.reduce((curr, next) => curr + next)}</div>;
    const wrapper = shallow(<FilterTag {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call onClear when closing tag when selected is set', () => {
    props.selected = ['selected value'];
    const wrapper = shallow(<FilterTag {...props} />);
    const tag = wrapper.find(Tag);
    tag.simulate('close');
    expect(props.onClear.called).toBeTruthy();
  });
  it('should not call onClear when closing tag when selected not set', () => {
    const wrapper = shallow(<FilterTag {...props} />);
    const tag = wrapper.find(Tag);
    tag.simulate('close');
    expect(props.onClear.called).toBeFalsy();
  });
});
