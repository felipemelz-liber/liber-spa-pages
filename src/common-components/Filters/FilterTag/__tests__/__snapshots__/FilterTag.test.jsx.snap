// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FilterTag Component should render correctly open 1`] = `
<Styled(ModalTag)
  label="label"
  show={true}
/>
`;

exports[`Testing FilterTag Component should render correctly with custom seleted Formatter 1`] = `
<Styled(ModalTag)
  closable={true}
  gray={true}
  label={
    <styled.div>
      <styled.div
        switch={false}
      >
        <styled.span>
          <div>
            6
          </div>
        </styled.span>
        <styled.span>
          label
        </styled.span>
      </styled.div>
    </styled.div>
  }
  onClose={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  show={false}
/>
`;

exports[`Testing FilterTag Component should render correctly with props 1`] = `
<Styled(ModalTag)
  label="label"
  show={false}
/>
`;

exports[`Testing FilterTag Component should render correctly with selected 1`] = `
<Styled(ModalTag)
  closable={true}
  gray={true}
  label={
    <styled.div>
      <styled.div
        switch={false}
      >
        <styled.span>
          selected value
        </styled.span>
        <styled.span>
          label
        </styled.span>
      </styled.div>
    </styled.div>
  }
  onClose={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  show={false}
/>
`;

exports[`Testing FilterTag Component should render correctly with selected wile open 1`] = `
<Styled(ModalTag)
  closable={true}
  gray={false}
  label={
    <styled.div>
      <styled.div
        switch={true}
      >
        <styled.span>
          selected value
        </styled.span>
        <styled.span>
          label
        </styled.span>
      </styled.div>
    </styled.div>
  }
  onClose={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  show={true}
/>
`;

exports[`Testing FilterTag Component should render correctly without props 1`] = `
<Styled(ModalTag)
  label=""
  show={false}
/>
`;
